bitcoin/pullpush.c

```diff
@@ -58,7 +58,7 @@ const u8 *pull(const u8 **cursor, size_t *max, void *copy, size_t n)
 	}
 	*cursor += n;
 	*max -= n;
-	if (copy)
+	if (copy && p)
```

Comment:
Would rather test `n` here so we'll get a nice crash if `p` is NULL somehow (shouldn't happen: *max will be 0 in that case, so `n` must be zero).